python/ray/data/tests/test_dataset_tfrecords.py

```diff
@@ -213,6 +214,16 @@ def test_write_invalid_tfrecords(ray_start_regular_shared, tmp_path):
         ds.write_tfrecords(tmp_path)
 
 
+def test_read_invalid_tfrecords(ray_start_regular_shared, tmp_path):
+    file_path = os.path.join(tmp_path, "file.json")
+    with open(file_path, "w") as file:
+        json.dump({"number": 0, "string": "foo"}, file)
+
+    # Expect exception raised when reading JSON as TFRecord file.
+    with pytest.raises(Exception, match="Failed to read TFRecord file"):
+        ray.data.read_tfrecords(file_path)
```

Comment:
Why is the `.json` file being read at all? Like, shouldn't this be filtered out by the file extension filter?

```
@PublicAPI(stability="alpha")
class TFRecordDatasource(FileBasedDatasource):

    _FILE_EXTENSION = "tfrecords"  # <--

    ....
```

Reply:
This file extension is only used when writing TFRecord file. We don't filter out file by ".tfrecords" when reading TFRecord file ([code](https://github.com/ray-project/ray/blob/master/python/ray/data/read_api.py#L1007)). I feel this is right choice so far, we are seeing user's TFRecord file has all different extensions/naming. They don't usually end with ".tfrecords".

Reply:
> This file extension is only used when writing TFRecord file. We don't filter out file by ".tfrecords" when reading TFRecord file ([code](https://github.com/ray-project/ray/blob/master/python/ray/data/read_api.py#L1007)).

Oh, didn't realize that. 

> I feel this is right choice so far, we are seeing user's TFRecord file has all different extensions/naming. 

Makes sense. OOC, what other extensions are common?

Reply:
I am seeing file extensions like ".tfrec", ".tfrecord", and not have extension at all.